rllib/examples/gpus/float16_training_and_inference.py

```diff
@@ -0,0 +1,241 @@
+"""Example of using float16 precision for training and inference.
+
+This example:
+    - shows how to write a custom callback for RLlib to convert all RLModules
```

Comment:
Absolutely awesome!!! This is such a great example!

Reply:
Thanks a ton!! Yes, and it actually works :D (only with adding all these weird tricks to stabilize learning). But I agree, it's super nice to see how users can customize not just one component at the same time, but several and then achieve a very high degree of customizability w/o us having to add feature after feature to the lib.